src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,16 +1,264 @@
-import { Component } from 'react';
+import classNames from 'classnames';
+import React, { useState } from 'react';
 
-type Props = {};
-type State = {};
+type Props = {
+  onAdd: (newMovie: Movie) => void;
+};
 
-export class NewMovie extends Component<Props, State> {
-  state: State = {};
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [newMovie, setNewMovie] = useState({
+    title: '',
+    description: 'No description',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
 
-  render() {
-    return (
-      <form>
-        Put the form here
-      </form>
+  const {
+    title,
+    imgUrl,
+    imdbUrl,
+    imdbId,
+  } = newMovie;
+  const validUrl = new RegExp(
+    // eslint-disable-next-line max-len
+    /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@.\w_]*)#?(?:[.!/\\\w]*))?)$/,
+  );
+  const validation = (
+    prevClass: string,
+    input: string,
+    urlQuery = false,
+  ) => {
+    if (urlQuery) {
+      const isValidUrl = validUrl.test(input);
+
+      return classNames(
+        prevClass,
+        { 'is-danger': !isValidUrl },
+        { 'is-success': isValidUrl },
+      );
+    }
+
+    return classNames(
+      prevClass,
+      { 'is-danger': !input },
+      { 'is-success': input },
     );
```

Comment:
```suggestion
  const validation = (
    prevClass: string,
    input: string,
    isUrlQuery = false,
  ) => {
    let isValid = !!input;
    
    if (isUrlQuery) {
      isValid = validUrl.test(input);
    }

    return classNames(
      prevClass,
      { 'is-danger': !isValid },
      { 'is-success': isValid },
    );
```
In both cases you have similar outcome, DRY